### DISCUSSION TITLE
Problem with column format in "empty" excel

### DISCUSSION THREAD

**Discussion by tasmanero** [ClosedXML/ClosedXML#2010] [Q&A]
I am creating through server code an "empty" excel that has a header with the names of the fields. The rest of the cells are empty but I want each column to have a format.
In the example that I send, for example, column A I want it to be text and column B a number with decimals and a comma, the general rest.
![image](https://user-images.githubusercontent.com/87016898/218514488-fcf3848d-58da-4c86-849e-60b74c5aad8c.png)

**Comment by jahav** [comment]
Columns have a Style property that can be used to set number format. If nothing is set, genernal format is used.  You can use predefined format or a custom format.
```
	var wb = new XLWorkbook();
	var ws = wb.AddWorksheet();
	ws.Column(1).Style.NumberFormat.SetNumberFormatId((int)XLPredefinedFormat.Number.Text);
	ws.Column(2).Style.NumberFormat.SetNumberFormatId((int)XLPredefinedFormat.Number.Precision2WithSeparator);

	// Note that the thousands groups and decimal separater will change with the culture.
	ws.Column(3).Style.NumberFormat.SetFormat("#,##0.00");

	ws.Cell(1, 3).Value = 12345.67;
	wb.SaveAs(@"c:\temp\format.xlsx");
```

For more info, see https://closedxml.readthedocs.io/en/latest/features/cell-format.html#number-format
